test_utils.go

```diff
@@ -32,24 +32,59 @@ import (
 )
 
 // fake stats
-type dummmyStats struct {
-	vals map[string]int64
+type dummyStats struct {
+	sync.RWMutex
+	_vals map[string]int64
 }
 
-func newDummyStats() *dummmyStats {
-	return &dummmyStats{make(map[string]int64)}
+func newDummyStats() *dummyStats {
+	return &dummyStats{
+		_vals: make(map[string]int64),
+	}
+}
+
+func (s *dummyStats) IncCounter(key string, tags bark.Tags, val int64) {
+	s.Lock()
+	defer s.Unlock()
+
+	s._vals[key] += val
+}
+
+func (s *dummyStats) read(key string) int64 {
+	s.Lock()
```

Comment:
There is no reason for performance optimizations in this specific code, but I will acquire a read lock here for good practice.